examples/CMakeLists.txt

```diff
@@ -35,6 +35,16 @@ IF(NOT TARGET DD4hep::DDCore)
 ENDIF()
 include(DD4hepMacros) 
 
+IF( "$ENV{DD4hepINSTALL}" STREQUAL "" )
+  IF( DD4hep_FOUND )
+    MESSAGE( FATAL_ERROR "Must source <DD4hep-install-dir>/bin/thisdd4hep.sh first" )
+    EXIT(1)
+  ENDIF()
+  # When building the examples together with DD4hep (-DDD4HEP_BUILD_EXAMPLES=ON),
+  # thisdd4hep.sh hasn't run. Manually set up the required variables for the tests.
+  SET( ENV{DD4hepINSTALL} ${CMAKE_INSTALL_PREFIX} )
```

Comment:
Does this actually do anything for the tests? It sets the environment variable for the cmake run, but when running ctest this isn't set any longer, is it?

Reply:
I see that some test definitions use ENV{DD4hepInstall}, can't this be DD4hep_ROOT?

Reply:
The environment variable is used in some places in the CMake code. For example, in examples/CLICSiD/CMakeLists.txt:

```cmake
  COMMAND    "${CMAKE_INSTALL_PREFIX}/bin/run_test_CLICSiD.sh"
  EXEC_ARGS  python ${DD4hep_ROOT}/bin/checkGeometry --compact=file:$ENV{DD4hepINSTALL}/DDDetectors/compact/SiD.xml
```

When cmake runs, `$ENV{DD4hepInstall}` will be evaluated, and the references to it disappear from the Makefiles, so it doesn't have to be set later on when running ctest. The tests set up the required environment again from scratch in their respective setup scripts. I think they only care about `$DD4hep_DIR` at runtime.

I'm not sure `$ENV{DD4hepInstall}` can be replaced with `${DD4hep_ROOT}` since, at the moment, you also want to be able to build the examples separately in a second build step (and possibly install them somewhere else). In that scenario, `${DD4hep_ROOT}` is not set and cmake picks up the location of the main DD4hep installation from `$ENV{DD4hepInstall}`. In fact, that's how the example tests are currently built in the CI workflow.

Reply:
`${DD4hep_ROOT} is set in DD4hepConfig.cmake
https://github.com/AIDASoft/DD4hep/blob/1b66f00c4793677f621bf21a812afb7af2a7b5e1/cmake/DD4hepConfig.cmake.in#L25
So if the examples are based on an existing installation, DD4hep_ROOT will also be available.

Reply:
I see. Well, then the detour through the environment can be avoided. Much cleaner indeed. Let me make the change and try it out.

Reply:
To be clear: I propose these changes

```diff
-  EXEC_ARGS  python ${DD4hep_ROOT}/bin/checkGeometry --compact=file:$ENV{DD4hepINSTALL}/DDDetectors/compact/SiD.xml
+  EXEC_ARGS  python ${DD4hep_ROOT}/bin/checkGeometry --compact=file:${DD4hep_ROOT}/DDDetectors/compact/SiD.xml
```

Reply:
Yup, exactly what I had in mind.

Reply:
Thanks!